### ISSUE TITLE
Ckan Harvester does not delete resources

### ISSUE THREAD

**Issue by bunnis** [ckan/ckanext-harvest#256]
Hi
Deleting all datasets harvested with the 'Clear' button on the admin page does not erase the resource files. I tried to add the following code on https://github.com/ckan/ckanext-harvest/blob/master/ckanext/harvest/logic/action/update.py#L105

The problem is that this takes too long and triggers a package_update (called from resource_delete). Also, since I have a lots of resources (5k +) I'm not sure they are getting deleted from the filesystem even with my method. This may be the right way of doing it but I think the correct way should be something that triggers the file deletion.

```
    # Clear all resources associated with the datasets
    q='harvest_source_id:'+str(harvest_source_id)
    
    i=0
    results_source_id = toolkit.get_action('package_search')(context, {'q':q,'rows':20000})
    print "starting deleting resources"

        
    for result_del in results_source_id['results']:
        #print i
        for resource_del in result_del['resources']:
            toolkit.get_action('resource_delete')(context, {'id':resource_del['id']})
        i=i+1
    print "done deleting resources"
```

**Comment by bunnis** [comment]
Another though, maybe, instead of doing resource_delete I should be doing dataset_purge? But then how does this handles the table entries?

**Comment by metaodi** [comment]
@bunnis good point. Please be aware that `resource_delete` does **not** delete the file in the FileStore. In order to do that, you need to call `resource_update` with the `clear` parameter.

**Comment by bunnis** [comment]
Can I use that parameter on the resource_delete call?

Does the dataset_purge include the deletion of the resource in the FileStore as default? I did some experiment and it seemed the case

**Comment by metaodi** [comment]
I have no experience with `dataset_purge`, but I can't find any reference to the FileStore in the code. And no, there is no `clear` parameter for `resource_delete`, so you first have to update the resource, then delete it.

**Comment by bunnis** [comment]
Thanks. I'll update the code and maybe do a PR

**Comment by bunnis** [comment]
Takes too long if the harvest source has lots of datasets. Maybe this request should be sent to the redis queue and handle it with a harvester run command?
